webapp/app/utils.py

```diff
@@ -12,4 +15,13 @@ def lru_cached(func):
     if Config.USE_LRU_CACHE:
         return lru_cache(func)
 
-    return func
\ No newline at end of file
+    return func
+
+
+def non_production_environment_required(f):
+    if os.environ.get('FLASK_ENV') != "production":
+        return f
+    else:
+        def disable_route(*args, **kwargs):
+            abort(404)
+        return disable_route
```

Comment:
I was thinking about another way to implement this decorator:
```python
def non_production_environment_required(f):
    def decorated(*args, **kwargs):
        if Config.ALLOW_TESTING_ROUTES:
            return f(*args, **kwargs)
        else:
            abort(404)

    return decorated
```

Basically the difference is that this returns a `decorated` function that is then later used and also set as a view function to the app. This would mean that if we use the decorator more than once all those functions would have the same name.

However, the current implementation will only call the decorator once (when `register_request_handlers(app)` is called). I think that this should not be a big problem because we do not change configuration values during runtime except for tests. Nevertheless, I would be very happy to hear your thoughts on that.

Reply:
Okay, @SannyNguyenHung had the idea to split it more up. Therefore, the decorator function is not used for every individual route but for a general testing routes register function. I had to change it back to way described above.  Especially because we do not set the view_function as `decorated` anymore.

I will keep that comment for reference. If you still have comments about it, I am very happy to hear your thoughts